bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -0,0 +1,286 @@
+            + conf.get(SNAPSHOT_RESTOREDIR_KEY)
+            + "\n"
+            + CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY
+            + " = "
+            + conf.get(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY)
+            + "\n");
+
+    Job job = createSubmittableJob(conf);
+    int successStatus = (job.waitForCompletion(true) ? 0 : 1);
+
+    Counters counters = job.getCounters();
+    long numRows = counters.findCounter(ScanCounter.NUM_ROWS).getValue();
+    long numCells = counters.findCounter(ScanCounter.NUM_CELLS).getValue();
+
+    System.out.println(("############# Num of ROWS imported= " + numRows));
+    System.out.println(("############# Num of CELLS imported= " + numCells));
+
+    return successStatus;
+  }
+
+  protected static void setConfFromArgs(Configuration conf, String[] args) {
+    // set arguments in configuration for ease of use
+    conf.set(SNAPSHOTNAME_KEY, args[0]);
+    conf.set(TABLENAME_KEY, args[1]);
+    conf.set(HConstants.HBASE_DIR, args[2]);
+    conf.set(SNAPSHOT_RESTOREDIR_KEY, args[3]);
+
+    // set default fs
+    conf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, conf.get(HConstants.HBASE_DIR));
+
+    // default job name
+    conf.setIfUnset(
+        IMPORT_SNAPSHOT_JOBNAME_KEY,
+        ImportHBaseSnapshotJob.class.getName()
+            + " - from snapshot "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + " to "
+            + conf.get(TABLENAME_KEY));
+
+    // implicit bigtable configs
+    conf.setIfUnset(
+        ClusterConnection.HBASE_CLIENT_CONNECTION_IMPL,
+        BigtableConfiguration.getConnectionClass().getName());
+    conf.setBooleanIfUnset(BigtableOptionsFactory.BIGTABLE_USE_TIMEOUTS_KEY, true);
+    conf.setIfUnset(
+        BigtableOptionsFactory.BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
+        String.valueOf(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_DEFAULT));
+
+    // implicit table outputformat configs
+    conf.set(TableOutputFormat.OUTPUT_TABLE, conf.get(TABLENAME_KEY));
+    conf.setStrings(
+        "io.serializations",
+        conf.get("io.serializations"),
+        MutationSerialization.class.getName(),
+        ResultSerialization.class.getName());
+  }
+
+  /**
+   * Sets up the actual job.
+   *
+   * @param conf The current configuration.
+   * @return The newly created job.
+   * @throws java.io.IOException When setting up the job fails.
+   */
+  protected static Job createSubmittableJob(Configuration conf) throws IOException {
+    Job job = Job.getInstance(conf, conf.get(IMPORT_SNAPSHOT_JOBNAME_KEY));
+    job.setJarByClass(ImportHBaseSnapshotJob.class);
+    TableMapReduceUtil.initTableSnapshotMapperJob(
+        conf.get(SNAPSHOTNAME_KEY),
+        new Scan().setMaxVersions(),
+        SnapshotMapper.class,
+        ImmutableBytesWritable.class,
+        Writable.class,
+        job,
+        true,
+        new Path(conf.get(SNAPSHOT_RESTOREDIR_KEY)));
+
+    job.setNumReduceTasks(0);
+    job.setOutputFormatClass(TableOutputFormat.class);
```

Comment:
Something needs to randomize the splits otherwise you'll sequentially hotspot each tablet

Reply:
hotspotting is targeted for a subsequent release due to complexity. however, in this release, we could add `BIGTABLE_BUFFERED_MUTATOR_ENABLE_THROTTLING` with a default low target that has helped reduce effects of hotspotting in our production use. thoughts in adding it as a default here with a 100ms target?

Reply:
I think you can expose latency based throttling as an option, but I dont think its relevant for my concern. Not randomizing the chunks will not allow this job to scale to multiple nodes.

Can you clarify where the complexity is coming from? Can't we use the same approach as ShuffledSequenceFileInputFormat?

Reply:
shuffled splits has been added and note in readme for indicating how to enable latency based throttling